x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/QueryConfig.java

```diff
@@ -124,4 +142,37 @@ public ActionRequestValidationException validate(ActionRequestValidationExceptio
         }
         return validationException;
     }
+
+    public void checkForDeprecations(String id, NamedXContentRegistry namedXContentRegistry, Consumer<DeprecationIssue> onDeprecation) {
+        LoggingDeprecationAccumulationHandler deprecationLogger = new LoggingDeprecationAccumulationHandler();
+
+        try {
+            queryFromXContent(source, namedXContentRegistry, deprecationLogger);
+        } catch (IOException e) {
+            onDeprecation.accept(
+                new DeprecationIssue(
+                    Level.CRITICAL,
+                    "Transform [" + id + "]: " + TransformMessages.LOG_TRANSFORM_CONFIGURATION_BAD_QUERY,
+                    TransformDeprecations.QUERY_BREAKING_CHANGES_URL,
+                    e.getMessage(),
+                    false,
+                    null
+                )
+            );
+        }
+
+        deprecationLogger.getDeprecations().forEach(deprecationMessage -> {
+            onDeprecation.accept(
+                new DeprecationIssue(
+                    Level.CRITICAL,
+                    "Transform [" + id + "] uses deprecated query options",
+                    TransformDeprecations.QUERY_BREAKING_CHANGES_URL,
+                    deprecationMessage,
+                    false,
+                    null
+                )
+            );
+        });
```

Comment:
ok, I like this way better than the way ML is doing it (these list of strings, etc.). Good refactoring I think++